packages/storybook/stories/wc-helpers.jsx

```diff
@@ -232,18 +246,18 @@ function CanvasLink() {
 export function StoryDocs({ componentName, data, children }) {
   const component = componentName || data?.tag;
   const componentDocs = additionalDocs?.[component];
-  const componentData = data ? { ...data, ...componentDocs } : null;
+  const componentData = data || componentDocs ? { ...data, ...componentDocs } : null;
+
+  const maturityScale = componentDocs?.maturityCategory ? componentDocs : getMaturityScale(data.docsTags);
 
-  const maturityCategory = componentDocs?.maturityCategory;
-  const maturityLevel = componentDocs?.maturityLevel;
+  const maturityCategory = maturityScale.category ?? componentDocs?.maturityCategory;
+  const maturityLevel = maturityScale.level ?? componentDocs?.maturityLevel;
   // This feels a bit awkward, but I didn't want to use a magic number
   const _componentName = data?.tag?.slice('va-'.length);
   // Default the guidance values to be based on the web component's
   // tag name where possible
-  const {
-    guidanceHref = _componentName,
-    guidanceName = capitalize(_componentName)?.replaceAll('-', ' '),
-  } = componentDocs;
+  const guidanceHref = getDocsTagValue('guidanceHref', data?.docsTags)[0] ?? (componentDocs?.guidanceHref ?? _componentName);
+  const guidanceName = componentDocs?.guidanceName ?? capitalize(_componentName)?.replaceAll('-', ' ') ?? componentName;
```

Comment:
Is the last `componentName` nullish coalescing necessary? I'm reading this as: If there isn't a `guidanceName` in the component's `additional-docs` object, then try to capitalize the web component's tag name, and if that isn't available then use the `componentName` prop. But they don't look like they would be part of a URL:

![ripgrep output for "componentName" on storybook story files. We see `componentName` props like "Modal - React" and "Icon - user"](https://user-images.githubusercontent.com/2008881/185972372-3f7f40cd-671a-4e46-a3c8-0bc2b87453ef.png)

Reply:
That's needed for the React components.

Reply:
I did some debugging and found that for a React component, `guidanceName` is defined, but `guidanceHref` isn't (one uses the undefined `_componentName` which may be an accident).

![Screenshot of the browser's debugger showing an undefined `guidanceHref` and a defined `guidanceName`. `componentDocs` is missing both properties.](https://user-images.githubusercontent.com/2008881/187497941-1384538b-b1f7-4de9-b84c-970c958cd676.png)

Since the href is undefined, this means we aren't seeing the `<Guidance>` component:

https://github.com/department-of-veterans-affairs/component-library/blob/8a145b1b5b66ff3009442555aab15c7a82eabb8a/packages/storybook/stories/wc-helpers.jsx#L267

https://github.com/department-of-veterans-affairs/component-library/blob/8a145b1b5b66ff3009442555aab15c7a82eabb8a/packages/storybook/stories/wc-helpers.jsx#L118-L119

If not seeing that is intentional, then I don't think we need the nullish coalescing.

But even if `guidanceHref` is defined so that it uses `componentName` instead of `_componentName`, like this:

```js
  const guidanceHref = getDocsTagValue('guidanceHref', data?.docsTags)[0] ?? (componentDocs?.guidanceHref ?? componentName);
```

Since the `guidanceHref` and `guidanceName` are defined, we see the `Guidance` component and it's using a broken URL:

![Screenshot of the Guidance component appearing and using a broken link based on the "Expanding group" `componentName`](https://user-images.githubusercontent.com/2008881/187500469-ce6c2148-b475-490f-b2d0-1cc9b6e723d9.png)

![Screenshot of a "Forbidden" error when clicking on the link provided in the `Guidance` component](https://user-images.githubusercontent.com/2008881/187501119-012e82ad-714a-4413-ab64-dee8e64c7b8e.png)

Reply:
I'm not sure I follow what you are doing there, but I went ahead and removed that last bit from the guidanceName and instead specified names for the two React components that needed it in the additional-docs file.